idaes_examples/notebooks/docs/unit_models/custom_unit_models/Aq_property.py

```diff
@@ -0,0 +1,306 @@
+                         should unfix any state variables fixed during
+                         initialization (default=False).
+                         True - states variables are not unfixed, and
+                         a dict of returned containing flags for
+                         which states were fixed during initialization.
+                         False - state variables are unfixed after
+                         initialization by calling the
+                         release_state method
+
+        Returns:
+            If hold_states is True, returns a dict containing flags for
+            which states were fixed during initialization.
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+
+        if state_vars_fixed is False:
+            # Fix state variables if not already fixed
+            flags = fix_state_vars(self, state_args)
+
+        else:
+            # Check when the state vars are fixed already result in dof 0
+            for k in self.keys():
+                if degrees_of_freedom(self[k]) != 0:
+                    raise Exception(
+                        "State vars fixed but degrees of freedom "
+                        "for state block is not zero during "
+                        "initialization."
+                    )
+
+        if state_vars_fixed is False:
+            if hold_state is True:
+                return flags
+            else:
+                self.release_state(flags)
+
+        init_log.info("Initialization Complete.")
+
+    def release_state(self, flags, outlvl=idaeslog.NOTSET):
+        """
+        Method to release state variables fixed during initialization.
+
+        Keyword Arguments:
+            flags : dict containing information of which state variables
+                    were fixed during initialization, and should now be
+                    unfixed. This dict is returned by initialize if
+                    hold_state=True.
+            outlvl : sets output level of logging
+        """
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="properties")
+
+        if flags is None:
+            return
+        # Unfix state variables
+        revert_state_vars(self, flags)
+        init_log.info("State Released.")
+
+
+@declare_process_block_class("AqPhaseStateBlock", block_class=_StateBlock)
+class AqPhaseStateBlockData(StateBlockData):
+    """
+    An example property package for ideal gas properties with Gibbs energy
+    """
+
+    def build(self):
+        """
+        Callable method for Block construction
+        """
+        super(AqPhaseStateBlockData, self).build()
+        self._make_state_vars()
+
+    def _make_state_vars(self):
+        self.flow_vol = Var(
+            initialize=1,
+            domain=NonNegativeReals,
+            doc="Total volumetric flowrate",
+            units=units.ml / units.min,
+        )
+
+        salts_conc = {"NaCl": 0.15, "KNO3": 0.2, "CaSO4": 0.1}
```

Comment:
This is not the way to do this. Firstly, `conc_mass_comp` should be indexed using the `component_list` defined in the parameter block, not by the keys of this dict. Secondly, you do not appear to be using the values in this dict to initialize the variable (you can send this dict directly to the `initialize` argument).

Reply:
I shall do this.